test/cql-pytest/test_unset.py

```diff
@@ -197,3 +202,24 @@ def test_unset_insert_where_lwt(cql, table2):
     stmt = cql.prepare(f'INSERT INTO {table2} (p, c) VALUES ({p}, ?) IF NOT EXISTS')
     with pytest.raises(InvalidRequest, match="unset"):
         cql.execute(stmt, [UNSET_VALUE])
+
+# Like test_unset_insert_where, but using UPDATE
+# Python driver doesn't allow sending an UNSET_VALUE for the partition key,
+# so only the clustering key is tested.
+def test_unset_update_where(cql, table3):
+    stmt = cql.prepare(f"UPDATE {table3} SET r = 42 WHERE p = 0 AND c = ?")
+
+    with pytest.raises(InvalidRequest, match="unset"):
+        cql.execute(stmt, [UNSET_VALUE])
+
+# Like test_unset_insert_where_lwt, but using UPDATE
+# Python driver doesn't allow sending an UNSET_VALUE for the partition key,
+# so only the clustering key is tested.
```

Comment:
Actually in addition to the clustering key, you're additionally checking an expression with a regular column, but what the heck ;-) I'll merge as-is.